examples/plot_binary_classification_COMPAS.py

```diff
@@ -0,0 +1,484 @@
+#
+# While this worked as the numbers show, it's not entirely obvious how it found
+# its solution. The following section will provide a deep dive on post
+# processing for Equalized Odds (EO).
+#
+# The post processing method (based on work by `Hardt, Price,
+# Srebro <https://arxiv.org/pdf/1610.02413.pdf>`_) takes a fairness-unaware model
+# and disparity constraints (such as EO) in the constructor and features (X),
+# labels (y), and a sensitive feature (sensitive_features) in the fit method.
+# It subsequently uses the model to make predictions for all samples in X. Note
+# that these predictions could be 0/1 (as in this example), or more categories,
+# or even real valued scores. In this case, this looks as follows:
+
+scores = estimator_wrapper.predict(X_train)
+scores
+
+###############################################################################
+# Finding threshold rules
+# ***********************
+#
+# The algorithm then tries to find all thresholding rules with which it could
+# divide the data. Any score for which the thresholding rule evaluates to true
+# is predicted to be 1. It does that for each group, so in this case for each
+# race. Depending on the unconstrained predictor's scores you could have lots
+# of thresholding rules, between each set of such values. For each rule we just
+# evaluate the following two probabilities empirically:
+#
+# .. math::
+#   P[\hat{Y} = 1 | Y = 0] \text{ which is labeled x below to indicate that it'll be plotted on the x-axis}
+#
+# .. math::
+#   P[\hat{Y} = 1 | Y = 1] \text{ which is labeled y below to indicate that it'll be plotted on the y-axis}
+#
+# The former is the false negative rate (of that group), while the latter is
+# the true positive rate (of that group). In this example, the threshold rules
+# would be the ones shown below:
+
+from fairlearn.postprocessing._threshold_optimizer import _reformat_and_group_data
+
+data_grouped_by_sensitive_feature = _reformat_and_group_data(
+    sensitive_features_train, y_train.astype(int), scores
+)
+data_grouped_by_sensitive_feature.describe()
+
+from fairlearn.postprocessing._roc_curve_utilities import _calculate_roc_points
+
+roc_points = {}
+for group_name, group in data_grouped_by_sensitive_feature:
+    roc_points[group_name] = _calculate_roc_points(
+        data_grouped_by_sensitive_feature.get_group(group_name), 0
+    )
+print("Thresholding rules:")
+roc_points
+
+###############################################################################
+# The base points with (x,y) as (0,0) and (1,1) always exist, because that
+# essentially just means that we're predicting everything as 0 or everything as
+# 1 regardless of the scores from the fairness-unaware model. Let's look at
+# both cases:
+#
+# - x=0, y=0, threshold rule ">inf": more than infinity is impossible, which
+#   means every sample is predicted as 0. That means :math:`P[\hat{Y} = 1 | Y = 0] = 0`
+#   (represented as x) because the predictions $\hat{Y}$ are never 1, and
```

Comment:
The math with $<math>$ doesn't seem to work. Probably needs to be replaced by :math:`<insert math here>`